src/components/Buttons/IconButton.tsx

```diff
@@ -32,20 +42,30 @@ export type ButtonIcon =
 type ButtonProps<T extends ElementType> = {
   variant: ButtonIcon;
   sizeInPx: number;
+  text?: { variant: TextVariant; color: Color; text: string };
 } & ComponentPropsWithoutRef<T>;
 
 const IconButton = <T extends ElementType = 'button'>({
   variant,
   sizeInPx,
+  text,
+  onClick,
   ...rest
 }: ButtonProps<T>): JSX.Element => {
   return (
-    <StyledIconButton
-      variant={variant}
-      size={sizeInPx}
-      {...rest}
-      aria-label={variant}
-    />
+    <Container onClick={onClick}>
+      <StyledIconButton
+        variant={variant}
+        size={sizeInPx}
+        {...rest}
+        aria-label={variant}
+      ></StyledIconButton>
+      {text && (
+        <Text variant={text.variant} color={text.color}>
+          {text.text}
+        </Text>
```

Comment:
I was not sure if its good idea to add text to IconButton or icon to TextButton, but I added optional text here in the end

Reply:
I added a Button, which accepts text, leftIcon, or rightIcon -prop. That one can still be improved I bet, but a better starting point for these kinds of buttons